cranelift/codegen/src/machinst/abi.rs

```diff
@@ -742,13 +752,13 @@ impl SigData {
 
     /// Get the number of arguments expected.
     pub fn num_args(&self) -> usize {
-        let len = self.arg_indices.end - self.arg_indices.start;
-        let len = usize::try_from(len).unwrap();
-        if self.stack_ret_arg.is_some() {
-            len - 1
+        let num = if self.stack_ret_arg.is_some() {
+            self.args_length - 1
         } else {
-            len
-        }
+            self.args_length
+        };
+
+        usize::try_from(num).unwrap()
```

Comment:
`usize::try_from` was necessary when `len` was a `u32`, because I guess Rust supports targets with 16-bit `usize`. But both here and in `num_rets` below, you can just use `usize::from` since these lengths are `u16` now.